### PR TITLE
Add statefulset/conditions ops for test helper

### PR DESCRIPTION
**Pull request by bogdando**
With that, other operators can refer those for providing
envtest coverage for statefulsets and conditions involving
test cases.

### CODE REVIEW

File: modules/test/helpers/conditions.go [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+	corev1 "k8s.io/api/core/v1"
+)
+
+type conditionsGetter interface {
+	GetConditions(name types.NamespacedName) condition.Conditions
+}
+
+type conditionGetterFunc func(name types.NamespacedName) condition.Conditions
+
+func (f conditionGetterFunc) GetConditions(name types.NamespacedName) condition.Conditions {
```
Comment (bogdando) [inline modules/test/helpers/conditions.go:30]: There are still *unused* problems with those. Quite a few operators copy and paste these. Let's figure out how to properly plug that into the test helper

**Comment by bogdando** [comment]
this is currently needed in order to align nova and placement tests, but other operators can benefit too

File: modules/test/helpers/conditions.go (hunk 1 above)
Comment (bogdando) [reply modules/test/helpers/conditions.go]: I'm a bit lost with adapting those as proper interfaces after moved from nova tests

File: modules/test/helpers/common.go [hunk 2]
```diff
@@ -19,17 +19,32 @@ import (
...
 )
 
+type conditionsGetter interface {
+	GetConditions(name types.NamespacedName) condition.Conditions
+}
+
+type ConditionGetterFunc func(name types.NamespacedName) condition.Conditions
+
+// ConditionGetterFunc -
+func (f ConditionGetterFunc) GetConditions(name types.NamespacedName) condition.Conditions {
```
Comment (bogdando) [inline modules/test/helpers/common.go:37]: with this, nova tests refactored can pass.
I cen see other operators use GetConditions() quite extensively, I believe this should help with testing expected conditions there as well

File: modules/test/helpers/conditions.go [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+		g.Expect(actualCondition.Reason).To(
+			t.Equal(expectedReason),
+			"%s condition has a different reason. Actual condition: %v", conditionType, actualCondition)
+		g.Expect(actualCondition.Message).To(
+			t.Equal(expecteMessage),
+			"%s condition has a different message. Actual condition: %v", conditionType, actualCondition)
+	}, tc.timeout, tc.interval).Should(t.Succeed())
+
+	tc.logger.Info("ExpectConditionWithDetails succeeded", "type", conditionType, "expected status", expectedStatus, "on", name)
+}
```
Comment (gibizer) [inline modules/test/helpers/conditions.go:77]: No newline at the end of the file. Interesting that it is not caught by go fmt or other lint

**Review by gibizer — CHANGES_REQUESTED** [review]
just two small nit but otherwise looks good

**Review by gibizer — APPROVED** [review]
/lgtm

**Pull request merged**